ludwig/trainers/trainer.py

```diff
@@ -453,6 +454,16 @@ def write_step_summary(cls, train_summary_writer, combined_loss, all_losses, ste
                 "cuda/total_memory_occupied", torch.cuda.mem_get_info()[1], global_step=step
             )
 
+        # Log nvidia-smi stats
```

Comment:
Since nvidia-smi is only relevant when using GPUs, let's fold this logging into the block above so that users only get these logs `if torch.cuda.is_available():`.